client.lua

```diff
@@ -935,28 +944,35 @@ local function IsWhitelistedWeaponStress(weapon)
     return false
 end
 
-local shootingSleep = 500
-CreateThread(function() -- Shooting
-    while true do
-        local isArmed = IsPedArmed(PlayerPedId(), 7)
-        if LocalPlayer.state.isLoggedIn and isArmed then
-            local ped = PlayerPedId()
-            local weapon = GetSelectedPedWeapon(ped)
-            if weapon ~= `WEAPON_UNARMED` then
-                if IsPedShooting(ped) and not IsWhitelistedWeaponStress(weapon) then
-                    if math.random() < config.StressChance then
-                        TriggerServerEvent('hud:server:GainStress', math.random(1, 3))
-                    end
+local hasWeapon = false
+
+local startWeaponStressThread = function(weapon)
```

Comment:
Use the function syntax instead of assigning a function to a variable to keep up with the style we use